--- opencv/3rdparty/libwebp/CMakeLists.txt	Sat Dec 20 11:30:09 2025
+++ opencv/3rdparty/libwebp/CMakeLists.txt	Sat Dec 27 22:14:15 2025
@@ -24,7 +24,11 @@
 if(WIN32)
   foreach(file ${lib_srcs})
     if("${file}" MATCHES "_avx2.c")
-      set_source_files_properties("${file}" COMPILE_FLAGS "/arch:AVX")
+      if(MSVC)
+        set_source_files_properties("${file}" COMPILE_FLAGS "/arch:AVX")
+      else()
+        set_source_files_properties("${file}" COMPILE_FLAGS "-mavx")
+      endif()
     endif()
   endforeach()
 endif()
